site/docs/skydoc_deprecation.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: default
+title: Skydoc deprecation
+stylesheet: docs
+---
+
+## Why was Skydoc deprecated?
+
+Skydoc functioned by evalutating Starlark files as if they were python. Unfortunately, while
+Starlark is **similar** to Python, there are some important syntactic differences between
+the languages. Assuming compatibility between the languages was inherently brittle, and resulted
+in a maintenance burden on the Starlark code. Specifically, if one of your transitive dependencies
+were to adopt a Starlark-compatible, python-incompatible construct, your Skydoc integration would
+break!
+
+Skydoc still exists as it's a nontrivial migration to Stardoc, but we plan on removing it
+in mid 2019.
+
+## How to migrate
+
+Stardoc is not a drop-in replacement for Skydoc. Its usage is slightly different, and it has some
+new features. It's recommended to take a look at the root Stardoc documentation, but here is
+a brief summary of some things to note for migration:
+
+### Docstring specification
+
+Stardoc uses inline documentation strings instead of python-style docstrings.
+For example, Skydoc documentation may have been specified with:
+
+```
+checkstyle = rule(
+    implementation = _impl,
+    attrs = {
+        "srcs": attr.label_list(allow_files = FileType([".java"]),
+        "deps": attr.label(),
+    },
+)
+"""Example rule documentation.
+
+Example:
+  Here is an example of how to use this rule.
+
+Args:
+  srcs: Source files used to build this target.
+  deps: Dependencies for this target.
+"""
+```
+
+...the equivalent for Stardoc is:
+
+```
+my_rule = rule(
+    implementation = _my_rule_impl,
+    doc = """
+Example rule documentation.
+
+Example:
+  Here is an example of how to use this rule.
+""",
+    attrs = {
+        "srcs" : attr.label_list(
+            doc = "Source files used to build this target.",
+        ),
+        "deps" : attr.label_list(
+            doc = "Dependencies for this target.",
+        ),
+    }
+)
+```
+
+### Different Starlark Rule
+
+Stardoc uses a different Starlark rule Skydoc with different attributes.
+
+See [Generating Documentation](https://skydoc.bazel.build/docs/generating_stardoc.html) for a
+tutorial on using the new rule, and the
+[Build Rule Reference](https://skydoc.bazel.build/docs/stardoc_reference.html) for information
+about the new `stardoc` rule itself.
+
+### Starlark Dependencies
+
+Stardoc depends on your `bzl` file, all of its dependencies, and all of its **transitive**
+dependencies, so that it can fully evaluate your Starlark code.
+`bazel-skylib`'s `bzl_library` is your "best friend" for tracking `bzl` dependencies.
+
+For example, suppose your `mypackage/foo.bzl` file depends on `other/package/bar.bzl`, which
+depends on `third/package/baz.bzl`.
+
+```
```

Comment:
Nit: Use a separate triple-quoted markdown block for each of these three files, to improve readability.

Reply:
Done, and also pulled the file names out.